From: Hector Palacios <hector.palacios@digi.com>
Date: Thu, 9 Oct 2025 13:16:00 +0200
Subject: [PATCH] ccimx95dvk: enable full access to certain regulators from
 non-secure LVM

The following PMIC regulators are used by the DVK:
- VDD_3V3 (SW1)
- VDD_1V8 (SW3)
- VDD_LDO1 (LDO1)
- VDD_SDIO2 (LDO2)

Upstream-Status: Inappropriate [DEY specific]

Signed-off-by: Hector Palacios <hector.palacios@digi.com>
---
 configs/ccimx95dvk.cfg           | 4 ++++
 configs/ccimx95dvk/config_scmi.h | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/configs/ccimx95dvk.cfg b/configs/ccimx95dvk.cfg
index c160dbce0b85..d7ee4d0115fa 100755
--- a/configs/ccimx95dvk.cfg
+++ b/configs/ccimx95dvk.cfg
@@ -554,6 +554,10 @@ BRD_SM_CTRL_TEST_A        ALL
 BRD_SM_SENSOR_TEMP_PF09   ALL
 BRD_SM_SENSOR_TEMP_PF5301 SET
 BRD_SM_SENSOR_TEMP_PF5302 SET
+BRD_SM_VOLT_SW1_VDD_3V3      ALL
+BRD_SM_VOLT_SW3_VDD_1V8      ALL
+BRD_SM_VOLT_LDO1_VDD_LDO1    ALL
+BRD_SM_VOLT_LDO2_VDD_SDIO2   ALL
 BUTTON                    ALL, test
 CLOCK_DISP1PIX            ALL
 CLOCK_DISP2PIX            ALL
diff --git a/configs/ccimx95dvk/config_scmi.h b/configs/ccimx95dvk/config_scmi.h
index 87de1fd591ca..83084a9d1fbd 100644
--- a/configs/ccimx95dvk/config_scmi.h
+++ b/configs/ccimx95dvk/config_scmi.h
@@ -601,6 +601,10 @@
         .sensorPerms[DEV_SM_SENSOR_TEMP_A55] = SM_SCMI_PERM_ALL, \
         .sensorPerms[DEV_SM_SENSOR_TEMP_ANA] = SM_SCMI_PERM_SET, \
         .sysPerms = SM_SCMI_PERM_NOTIFY, \
+        .voltPerms[BRD_SM_VOLT_LDO1_VDD_LDO1] = SM_SCMI_PERM_ALL, \
+        .voltPerms[BRD_SM_VOLT_LDO2_VDD_SDIO2] = SM_SCMI_PERM_ALL, \
+        .voltPerms[BRD_SM_VOLT_SW1_VDD_3V3] = SM_SCMI_PERM_ALL, \
+        .voltPerms[BRD_SM_VOLT_SW3_VDD_1V8] = SM_SCMI_PERM_ALL, \
     }
 
 /*! Config for SCMI channel 5 */
